### PR TITLE
feat: add webp image convert function

### LINKED ISSUE #1788: Feature Request: Support WebP Image Format
**环境信息**

go-cqhttp版本: v1.0.0-rc3

**需要添加的功能内容**

Currently PCQQ client don't support WebP format, any webp image became blank images in PCQQ (although it is okay on mobile side).
Is it possible to auto-transform webp images to png / jpeg at `go-cqhttp` side? May it be an extra switch in YAML configuration files corresponds this feature in my opinion.

目前 PCQQ 客户端不支持 WebP 格式，在 PCQQ 上任何 webp 图像变成空图片（尽管在移动端它是好的）。
是否可以在 `go-cqhttp` 方面自动转换 webp 图像到 png / jpeg？我觉得它可能作为一个额外的开关在 YAML 配置文件里对应这个功能。

------

cc @shigma

### PR DESCRIPTION
**Pull request by MaikoTan**
close #1788

### CODE REVIEW

File: coolq/bot.go [hunk 1]
```diff
@@ -153,9 +156,16 @@ func (bot *CQBot) uploadLocalImage(target message.Source, img *LocalImageElement
...
 	}
-	if mt, ok := mime.CheckImage(img.Stream); !ok {
+	mt, ok := mime.CheckImage(img.Stream)
+	if !ok {
 		return nil, errors.New("image type error: " + mt)
 	}
+	if mt == "image/webp" && base.ConvertWebpImage {
+		img0, _ := webp.Decode(img.Stream)
+		stream := bytes.NewBuffer(nil)
+		png.Encode(stream, img0)
```
Comment (Mrs4s) [inline coolq/bot.go:166]: 这地方的错误最好处理一下

**Review by Mrs4s — CHANGES_REQUESTED** [review]

File: coolq/bot.go (hunk 1 above)
Comment (MaikoTan) [reply coolq/bot.go:166]: > It's better to deal with the mistakes in this place

Sure.

**Review by Mrs4s — APPROVED** [review]

**Comment by MaikoTan** [comment]
Well, I didn't even learn how to fix the lint errors

**Comment by Mrs4s** [comment]
> Well, I didn't even learn how to fix the lint errors

lint errors are not caused by this pull request. will fix it later

**Pull request merged**